docs/about/release-notes/migration-faq.md

```diff
@@ -0,0 +1,237 @@
+```
+
+```{dropdown} Can different pipeline stages use different Python environments or dependencies?
+
+Yes, you can specify different Conda environments per stage. Ensure module imports occur inside the stage's process or setup methods (not at the global scope).
+```
+
+---
+
+## Data Processing, Models, and Quality
+
+```{dropdown} How do you handle quality filtering for low-resource languages where no good models exist?
+
+NeMo Curator suggests several strategies (see {ref}`Heuristic Filtering <text-process-data-filter-heuristic>` and {ref}`Quality Assessment <about-concepts-text-data-processing>` for details):
+
+- Use available multilingual models (for example, Qwen, Mistral, or other models with many language capabilities)
+- Annotate high-quality English data with a classifier, translate these data to the target language, and then train a smaller in-language model
+- Upsample high-quality data or downsample and rewrite lower-quality data instead of discarding it (using an LLM rewrite for quality improvement)
+```
+
+```{dropdown} What is the "ensemble model bucketing" workflow?
+
+A set of models assigns a quality score (for example, 0–20), bucketed into high, medium, or low groups. High-quality data might be upsampled or have synthetic variants generated; low-quality data is rewritten rather than simply dropped—especially valuable for low-resource contexts.
+```
+
+```{dropdown} How can we implement or reuse rule-based or custom quality filters? Is YAML (configuration-based) support available?
+
+Quality filters can be implemented as full Python stages in the pipeline (see {ref}`Heuristic Filtering <text-process-data-filter-heuristic>` for available filters and usage examples). YAML or configuration-based filter definitions are available, making it easier to define and reuse filters without writing as much code. Collaboration is encouraged—please contribute region or language-specific filters via pull requests.
+```
+
+```{dropdown} Can the system track the number of documents or tokens processed, filtered, or passed at each stage?
+
+Each task is a data class. You can add whatever statistics you need (input or output counts, tokens dropped, and so on) within stage logic for detailed reporting or logging.
+```
+
+```{dropdown} Does the new deduplication feature support global deduplication (across all snapshots, not just incremental)?
+
+Yes, the Ray-powered NeMo Curator supports massive-scale {ref}`global deduplication <text-process-data-dedup>` using efficient, GPU-accelerated MinHash or other methods. For comprehensive documentation, refer to {ref}`Deduplication Concepts <about-concepts-deduplication>`.
```

Comment:
Only minhash is supported for fuzzy, right? Or is this question supposed to be about all 3 of exact, fuzzy, and semantic deduplication?